net/unbound/files/unbound.sh

```diff
@@ -1047,9 +1026,28 @@ unbound_uci() {
 
 ##############################################################################
 
+unbound_ismain() # true=0, false=1.
+{
+  if [ "$UNBOUND_N_RX_PORT" != "53" ] ; then
```

Comment:
Oops, this back-to-back-testing requirement was not in my thougts.

In normal production use-cases it would be very unlikely that both dnsmasq and unbound would be configured both to own localhost:domain (it would be an outright configuration error). For that reason the test of the port number has been put first, as kind of sanity check.

After the port 53 test, unbound_ismain() still tests if dnsmasq is enabled (guarded with the condition that the dnsmasq init script is executable). Only the -x test on the dnsmasq binary has been dropped. That one seemed a little over the top for me, when the dnsmasq init script is in order, one should assume that the daemon itself is present.

In the original code the test for the port conflict was absent. It was added as part of the invariant.  So, is the re-ordering of the tests problematic or does the port-conflict-test conflict with your back-to-back testing ?